pavelib/servers.py

```diff
@@ -116,33 +164,66 @@ def celery(options):
 @needs('pavelib.prereqs.install_prereqs')
 @cmdopts([
     ("settings=", "s", "Django settings for both LMS and Studio"),
+    ("asset_settings=", "a", "Django settings for updating assets for both LMS and Studio (defaults to settings)"),
     ("worker_settings=", "w", "Celery worker Django settings"),
     ("fast", "f", "Skip updating assets"),
+    ("optimized", "f", "Run with optimized assets"),
     ("settings_lms=", "l", "Set LMS only, overriding the value from --settings (if provided)"),
+    ("asset_settings_lms=", "al", "Set LMS only, overriding the value from --asset_settings (if provided)"),
     ("settings_cms=", "c", "Set Studio only, overriding the value from --settings (if provided)"),
+    ("asset_settings_cms=", "ac", "Set Studio only, overriding the value from --asset_settings (if provided)"),
+    ("no_contracts", "f", "Disable contracts. By default, they're enabled in devstack."),
 ])
 def run_all_servers(options):
     """
     Runs Celery workers, Studio, and LMS.
     """
     settings = getattr(options, 'settings', DEFAULT_SETTINGS)
-    settings_lms = getattr(options, 'settings_lms', settings)
-    settings_cms = getattr(options, 'settings_cms', settings)
+    asset_settings = getattr(options, 'asset_settings', settings)
     worker_settings = getattr(options, 'worker_settings', 'dev_with_worker')
     fast = getattr(options, 'fast', False)
+    optimized = getattr(options, 'optimized', False)
+    no_contracts = getattr(options, 'no_contracts', False)
 
-    if not fast:
-        args = ['lms', '--settings={}'.format(settings_lms), '--skip-collect']
-        call_task('pavelib.assets.update_assets', args=args)
+    if optimized:
+        settings = OPTIMIZED_SETTINGS
+        asset_settings = OPTIMIZED_ASSETS_SETTINGS
 
-        args = ['studio', '--settings={}'.format(settings_cms), '--skip-collect']
-        call_task('pavelib.assets.update_assets', args=args)
+    settings_lms = getattr(options, 'settings_lms', settings)
+    settings_cms = getattr(options, 'settings_cms', settings)
+    asset_settings_lms = getattr(options, 'asset_settings_lms', asset_settings)
+    asset_settings_cms = getattr(options, 'asset_settings_cms', asset_settings)
+    collect_static = not fast and asset_settings != settings
 
+    if not fast:
+        args = [
+            'lms', 'studio',
+            '--settings={}'.format(asset_settings),
+            '--skip-collect'
```

Comment:
Yes. This one is a bit subtle, so it deserves a comment. I factored out the calls to collect assets because they are invoked with possibly different settings files. I first call update_assets while skipping collect so that all the other stuff is done (SASS, Coffee etc) once.